x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportPutShutdownNodeAction.java

```diff
@@ -49,8 +56,42 @@ protected void masterOperation(
         ClusterState state,
         ActionListener<AcknowledgedResponse> listener
     ) throws Exception {
-        // TODO: implement me!
-        listener.onResponse(AcknowledgedResponse.of(true));
+        // Verify that the requested node actually exists in the cluster
+        if (state.nodes().nodeExists(request.getNodeId()) == false) {
+            throw new IllegalArgumentException("there is no node with id [" + request.getNodeId() + "] in this cluster");
+        }
```

Comment:
I don't think we should have this constraint, for example, if you need to add a shutdown for a node that is having connectivity issues, we should still allow marking it as shut down so that data can be migrated, rather than playing whack-a-mole waiting for it to be part of the cluster so that it can be marked as shutting down.

What do you think?

Reply:
Good point, I wasn't thinking about that when I wrote this. It would be nice to have some kind of validation, so that if you completely fumble the node ID it doesn't just silently accept that.

Looking around at the code though, it doesn't look like there's much (any?) validation of node IDs - a lot of tests just use arbitrary strings, so maybe there's nothing we can really do here (besides maybe a length limit?).